docs/tutorials/intro.md

```diff
@@ -200,19 +204,54 @@ Here's an internationalized date!
+We're going to use the `icu_datagen` crate to do this. It pulls in [CLDR data](http://cldr.unicode.org/index/downloads) and Unicode property data from ICU4C releases to generate the data.
 
-The whole thing can be done by running:
+The `icu_datagen` crate has a command line interface as well as a Rust API, which can be used in `build.rs` files. Here we will use the CLI.
 
+```
+cargo run -p icu_datagen --features bin -- \
+    --cldr-tag latest \
+    --icuexport-tag latest \
+    --out my-data \
+    --format blob \
+    --all-keys --all-locales
+```
+
+The last command is a bit dense, so let's dissect it.
+
+* First, we call `cargo run` which runs a binary
+* We tell it that the crate is named `icu_datagen`
+* We use `--features bin` to be able to build the binary (otherwise the crate will only contain the library)
+* Then we use `--` to separate arguments to `cargo` from arguments to `icu_datagen`
+* `--cldr-tag` informs the program which CLDR version to use
+* `--icuexport-tag` informs the program which ICU-exported data version to use
+* `--out` is the location where we want the generated ICU4X data to be stored
+* `--format blob` to set the format of the output (we'll discuss formats later)
+* and finally `--all-keys --all-locales` to specify that we want to export all keys available, for all locales.
+
+This should generate a `my-data` file containing your data in serialized form.
+
+*Notice:* `--cldr-tag` and `--icuexport-tag` fetch the specified version from the network. For offline or unconventional use, the user can also pass `--cldr-root`/`--icuexport-root` to use local versions.
+
+*Notice:* `--all-keys --all-locales` is a lot of data, in many cases you probably only want the keys for a particular component, for a set of locales you plan to support in the application. Datagen has alternate flags like `--keys` or `--key-file` for this level of control.
 
-```sh
+*Notice:* This command builds in debug mode since it builds faster; but if you plan on running `icu_datagen` a lot (for example, if you wish to create multiple per-key blobs to selectively load), we highly recommend using `--release`.
+
+## Using the generated data
+
+Once you have generated your data, it needs to be loaded as a data provider. The blob format we chose can be loaded by `BlobDataProvider` from the `icu_provider_blob` crate.
+
+This provider performs deserialization, so it's a `BufferProvider`. This means that the feature `"serde"` needs to be enabled on `icu`.
+
+The whole thing can be done by running:
+
+```console
 $ cargo add icu --features serde
-$ cargo add icu_provider_fs
-$ cargo add icu_provider --features deserialize_json
+$ cargo add icu_provider_blob
```

Comment:
issue: you also need `--features deserialize_json` or `--features deserialize_postcard_1` with `cargo add icu_provider`

Reply:
Not with blob provider, that activates postcard itself.